src/connection_string.ts

```diff
@@ -588,6 +588,9 @@ export const OPTIONS = {
   autoEncryption: {
     type: 'record'
   },
+  bsonRegExp: {
```

Comment:
I'm not sure we want this to be a connection string option as it's Node driver specific.

Reply:
I included the new bsonRegExp option here because this is where all possible options (uri, object param, and defaults) are defined so it might be a little overreaching to solve this issue now. I can create a new ticket to make sure you can’t use it in the connection string if that sounds better?

Reply:
Capturing some investigation from a slack convo:
Andy discovered we do support BSON options in the URI in 3.6 so the behavior is consistent, may be unusual, not sure what other drivers do but I think we can certainly defer changing that to another ticket.

Reply:
I'm fine with that.